irods/__init__.py

```diff
@@ -1,3 +1,8 @@
+import sys
+if sys.version_info[0] < 3:
```

Comment:
let's also check minor version... aka that we're 3.8+.

Reply:
Have we settled on 3.8 as a minimum, then?  Sounds good to me.